src/components/v5/common/CompletedAction/partials/StreamingPayment/partials/CancelModal/CancelModal.tsx

```diff
@@ -0,0 +1,142 @@
+import { Prohibit } from '@phosphor-icons/react';
+import React, { type FC } from 'react';
+
+import { getRole } from '~constants/permissions.ts';
+import { useAppContext } from '~context/AppContext/AppContext.ts';
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import { ButtonWithLoader } from '~frame/Extensions/pages/ExtensionDetailsPage/partials/ExtensionDetailsHeader/ButtonWithLoader.tsx';
+import useAsyncFunction from '~hooks/useAsyncFunction.ts';
+import { ActionTypes } from '~redux';
+import { type CancelStreamingPaymentPayload } from '~redux/types/actions/expenditures.ts';
+import { Form } from '~shared/Fields/index.ts';
+import { getHighestTierRoleForUser } from '~transformers';
+import { extractColonyRoles } from '~utils/colonyRoles.ts';
+import { formatText } from '~utils/intl.ts';
+import DecisionMethodSelect from '~v5/common/CompletedAction/partials/PaymentBuilder/partials/DecisionMethodSelect/DecisionMethodSelect.tsx';
+import Button from '~v5/shared/Button/index.ts';
+import Modal from '~v5/shared/Modal/index.ts';
+
+import { cancelDecisionMethodItems, validationSchema } from './consts.ts';
+import { type CancelModalProps } from './types.ts';
+import { getCancelDecisionMethodDescriptions } from './utils.ts';
+
+const CancelModal: FC<CancelModalProps> = ({
+  isOpen,
+  onClose,
+  streamingPayment,
+  onSuccess,
+  ...rest
+}) => {
+  const { user } = useAppContext();
+  const { colony } = useColonyContext();
+
+  const cancel = useAsyncFunction({
+    submit: ActionTypes.STREAMING_PAYMENT_CANCEL,
+    error: ActionTypes.STREAMING_PAYMENT_CANCEL_ERROR,
+    success: ActionTypes.STREAMING_PAYMENT_CANCEL_SUCCESS,
+  });
+
+  const handleCancelStreamingPayment = async ({
+    shouldWaive,
+  }: Pick<CancelStreamingPaymentPayload, 'shouldWaive'>) => {
+    try {
+      if (!streamingPayment) {
+        return;
+      }
+
+      const payload: CancelStreamingPaymentPayload = {
+        colonyAddress: colony.colonyAddress,
+        streamingPayment,
+        userAddress: user?.walletAddress ?? '',
+        shouldWaive,
+      };
+
+      await cancel(payload);
+      onSuccess();
```

Comment:
I think we should refetch the streaming payment here or as part of the onSuccess handler so that we have the latest claim amount.